File: mobile/lib/models/listing_offer.dart
class ListingOffer {
  final int id;
  final int listingId;
  final int userId;
  final String username;
  final String? profileImageUrl;
  final double amount;
  final DateTime createdAt;

  const ListingOffer({
    required this.id,
    required this.listingId,
    required this.userId,
    required this.username,
    this.profileImageUrl,
    required this.amount,
    required this.createdAt,
  });

  factory ListingOffer.fromJson(Map<String, dynamic> j) {
    return ListingOffer(
      id: (j['id'] as int?) ?? 0,
      listingId: (j['listing_id'] as int?) ?? 0,
      userId: (j['user_id'] as int?) ?? 0,
      username: j['username'] as String? ?? '',
      profileImageUrl: j['profile_image_url'] as String?,
      amount: (j['amount'] as num?)?.toDouble() ?? 0.0,
      createdAt: DateTime.tryParse(j['created_at'] as String? ?? '') ?? DateTime.now(),
    );
  }
}

File: backend/app/models/listing_offer.py
from datetime import datetime

from sqlalchemy import Float, DateTime, ForeignKey, func, Index
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class ListingOffer(Base):
    __tablename__ = "listing_offers"
    __table_args__ = (
        Index("ix_listing_offers_listing_id", "listing_id"),
        Index("ix_listing_offers_user_id", "user_id"),
    )

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    listing_id: Mapped[int] = mapped_column(
        ForeignKey("listings.id", ondelete="CASCADE"), nullable=False
    )
    user_id: Mapped[int] = mapped_column(
        ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    amount: Mapped[float] = mapped_column(Float, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
